Classes/Private/Manager/Database/OCTDBCall.m

```diff
@@ -0,0 +1,18 @@
+//
+//  OCTDBCall.m
+//  objcTox
+//
+//  Created by Chuong Vu on 6/8/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "OCTDBCall.h"
+
+@implementation OCTDBCall
+
+- (BOOL)isEqual:(OCTDBCall *)object
+{
+    return self.chat.uniqueIdentifier == object.chat.uniqueIdentifier;
+}
```

Comment:
When you implement `isEqual:` method you __have to__ implement `hash` as well. You can read [this article](http://nshipster.com/equality/) about that.

Anyway, I don't see why we want `OCTDBCalls` to be equal that way. Also Realm provide equality methods for object with defined primary key.
That's why I propose you to add `uniqueIdentifier` back, but always generate it. Don't use chat or someone else identifier.

Also I think that it would be nice to add `uniqueIdentifier` to all objects, to be more consistent. I think I'll create some superclass with identifier and then refactor all DB classes.